docs/flow/tutorials/collecting-prometheus-metrics.md

```diff
@@ -0,0 +1,65 @@
+---
+aliases:
+- /docs/agent/latest/flow/tutorials/collecting-prometheus-metrics
+title: Collecting Prometheus metrics
+weight: 200
+---
+
+Grafana Agent is a telemetry collector and the primary goal is to get telemetry data from one location to another. The goal of this tutorial is to get a basic working example of Grafana Agent working in Flow mode.  
+
+# Prerequisites
+
+* [Docker](https://www.docker.com/products/docker-desktop)
+* Clone the [Agent Repository](https://github.com/grafana/agent) `git clone git@github.com:grafana/agent.git`
+
+# Scraping the Agent
+
+To quickly spin up an example environment run the following: `docker-compose -f ./assets/scraping-the-agent.yaml` with the [flow file](../assets/flow_configs/agent.flow). Allow the service to run for a few minutes and you should be able to navigate to the [metrics browser](http://localhost:3000/explore?orgId=1&left=%5B%22now-1h%22,%22now%22,%22Cortex%22,%7B%22refId%22:%22A%22,%22instant%22:true,%22range%22:true,%22exemplar%22:false,%22expr%22:%22agent_build_info%22%7D%5D). 
+
+![](./assets/agent_build_info.png)
+
+This example scrapes the Grafana Agent's `http://localhost:12345/metrics` endpoint and pushes those metrics to the cortex instance. 
+
+
+## Scraping Component
+
+The [`prometheus.scrape`]({{< relref "prometheus.scrape.md" >}}) component is responsible for scraping the metrics of a particular endpoint and passing them on to another component.
+
+```river
+// prometheus.scrape is the type of component and "default" is the name or label of a component
+prometheus.scrape "default" {
+    // Tell the scraper to scrap at http://localhost:12345/metrics
+    // the http:// and metrics are implied but able to be overwritten 
+    targets = [{"__address__" = "localhost:12345"}]
+    // Forward the scrape results to the receiver, in general 
+    // Flow uses forward_to to tell which receiver to send results to. 
+    // The forward_to is an argument of prometheus.scrape.default  and 
+    // the receiver is an export prometheus.remote_write.prom
+    forward_to = [prometheus.remote_write.prom.receiver]
+}
+```
+
+The `prometheus.scrape "default"` annotation tells the type `prometheus.scrape` and the label `default`. All components must have a unique combination of type and name.
```

Comment:
We're using three terms here, 'type', 'label' and 'name'. I think I've used similar terms on the River docs and would ideally like to converge on a common appproach.

My 2c is that we can refer to 
- component 'type' = namespace (we only have Prometheus, discovery and local components now)
- component 'name' = the block name you have to pass, in this case `prometheus.scrape`
- component 'label' = the block label, here "default"

How does that sound to you?

Reply:
I'd suggest we drop it down to two terms:

* Component name: the name of the component `prometheus.scrape`, `prometheus.integration.node_exporter` 
* Component label: the user-assigned label for the component, only provided when the component requires a label. 

I'm not sure if the "namespace" idea is something that becomes user facing. At the moment it seems like an abstract concept we use for naming similar components, but isn't a strict thing users need to be aware of.

Reply:
Sounds good to me as well. If Matt is also on board, I'll check the other River docs to make sure we're aligned.

Reply:
Sounds good to me, will change,